lib/pg_ha_migrations/unsafe_statements.rb

```diff
@@ -115,7 +136,69 @@ def unsafe_add_index(table, column_names, options = {})
 
     options[:name] = validated_index.name
 
-    execute_ancestor_statement(:add_index, table, column_names, **options)
+    if options[:algorithm] == :concurrently
+      execute_ancestor_statement(:add_index, table, column_names, **options)
+    else
+      safely_acquire_lock_for_table(table, mode: :share) do
+        execute_ancestor_statement(:add_index, table, column_names, **options)
+      end
+    end
+  end
+
+  def unsafe_remove_index(table, column = nil, **options)
+    if options[:algorithm]== :concurrently
+      execute_ancestor_statement(:remove_index, table, column, **options)
+    else
+      safely_acquire_lock_for_table(table) do
+        execute_ancestor_statement(:remove_index, table, column, **options)
+      end
+    end
+  end
+
+  def unsafe_add_foreign_key(from_table, to_table, **options)
+    safely_acquire_lock_for_table(from_table, to_table, mode: :share_row_exclusive) do
+      execute_ancestor_statement(:add_foreign_key, from_table, to_table, **options)
+    end
+  end
+
+  def unsafe_remove_foreign_key(from_table, to_table = nil, **options)
+    calculated_to_table = options.fetch(:to_table, to_table)
+
+    if calculated_to_table.nil?
+      raise PgHaMigrations::InvalidMigrationError.new("The :to_table positional arg / kwarg is required for lock acquisition")
+    end
+
+    safely_acquire_lock_for_table(from_table, calculated_to_table) do
+      execute_ancestor_statement(:remove_foreign_key, from_table, to_table, **options)
+    end
+  end
+
+  def unsafe_rename_enum_value(name, old_value, new_value)
+    if ActiveRecord::Base.connection.postgresql_version < 10_00_00
+      raise PgHaMigrations::InvalidMigrationError, "Renaming an enum value is not supported on Postgres databases before version 10"
+    end
+
+    raw_execute("ALTER TYPE #{PG::Connection.quote_ident(name.to_s)} RENAME VALUE '#{PG::Connection.escape_string(old_value)}' TO '#{PG::Connection.escape_string(new_value)}'")
+  end
+
+  def unsafe_make_column_not_nullable(table, column, **options) # options arg is only present for backwards compatiblity
+    safely_acquire_lock_for_table(table) do
+      raw_execute("ALTER TABLE #{table} ALTER COLUMN #{column} SET NOT NULL") # TODO: quoting
```

Comment:
Is the TODO here new?

Reply:
It was, but I pushed up this commit to address (that was fixed in a previous PR and I messed up the rebase)

https://github.com/braintree/pg_ha_migrations/pull/113/commits/9adbc2bfda45636ac7aeb2335ed53bff3aac3261